src/lightning/pytorch/trainer/connectors/accelerator_connector.py

```diff
@@ -182,7 +183,7 @@ def _check_config_and_set_final_flags(
 
         """
         if plugins is not None:
-            plugins = [plugins] if not isinstance(plugins, list) else plugins
+            plugins = [plugins] if not isinstance(plugins, Iterable) else plugins
```

Comment:
@jedyang97 I'm ok with it either way. However, if this PR should land then we would want to update the type in the signature. And apply the same changes to fabric/connector.py as well. 

On the other hand, we have arguments in other places where the user can pass in a list or tuple of elements, and nothing would prevent it to be a general Iterable but we still prefer to advertise the simple types just because of simplicity so it's clear to the user what they can pass in.

Reply:
Addressed in #20452